bind/gengo.go

```diff
@@ -752,6 +752,97 @@ func (g *goGen) genType(sym *symbol) {
 		g.Printf("}\n\n")
 	}
 
+	if sym.isMap() {
+		ktyp := sym.GoType().Underlying().(*types.Map).Key()
```

Comment:
this whole block (inside the `if sym.isMap()`) could be factored inside a `func (g *goGen) genTypeTPMapping(sym *symbol)` method.

so this `genType()` method would look like:

```go
g.genTypeTPMapping(sym)
g.genTypeTPCall(sym)
```

and the first line of `genTypeTPMapping` would look like:
```go
func (g *goGen) genTypeTPMapping(sym *symbol) {
    if !sym.isMap() {
        return
    }
    // ...
}
```